test/unit/specs/vislib/visualizations/tile_map.js

```diff
@@ -196,6 +199,229 @@ define(function (require) {
+          vis.handler.charts.forEach(function (chart) {
+            chart._attr.heatNormalizeData = true;
+            var mapData = chart.chartData.geoJson;
+            var max = mapData.properties.allmax;
+            var i = _.random(0, mapData.features.length - 1);
+            var feature = mapData.features[i];
+            var lat = feature.geometry.coordinates[1];
+            var lng = feature.geometry.coordinates[0];
+            var intensity = parseInt(feature.properties.count / max * 100);
+            var array = chart.dataToHeatArray(mapData, max);
+            expect(array[i][0]).to.be(lat);
+            expect(array[i][1]).to.be(lng);
+            expect(array[i][2]).to.be(intensity);
+          });
+        });
+
+      });
+
+      describe('applyShadingStyle method', function () {
+        it('should return an object', function () {
+          vis.handler.charts.forEach(function (chart) {
+            var mapData = chart.chartData.geoJson;
+            var i = _.random(0, mapData.features.length - 1);
+            var feature = mapData.features[i];
+            var min = mapData.properties.allmin;
+            var max = mapData.properties.allmax;
+            expect(_.isObject(chart.applyShadingStyle(feature, min, max))).to.be(true);
+          });
+        });
+      });
+
+      describe('getBounds method', function () {
+        it('should return bounds object that contains every point in mapData', function () {
+          vis.handler.charts.forEach(function (chart) {
+            var mapData = chart.chartData.geoJson;
+            var containsPoint = [];
+            var bounds = chart.getBounds(mapData);
+            for (var i = 0; i < mapData.features.length; i++) {
+              var pt = [mapData.features[i].geometry.coordinates[1], mapData.features[i].geometry.coordinates[0]];
+              containsPoint.push(bounds.contains(pt));
+            }
+            containsPoint = _.compact(containsPoint);
+            expect(containsPoint.length).to.be(mapData.features.length);
+          });
+        });
+      });
+
+      describe('showTooltip method', function () {
+        it('should create a .leaflet-popup-kibana div for the tooltip', function () {
+          vis.handler.charts.forEach(function (chart) {
+            chart.tooltipFormatter = function (str) {
+              return str;
+            };
+            var map = chart.maps[0];
+            var mapData = chart.chartData.geoJson;
+            var layerIds = _.keys(map._layers);
+            var id = layerIds[_.random(1, layerIds.length - 1)]; // layer 0 is tileLayer
+            map._layers[id].fire('mouseover');
+            expect($(vis.el).find('.leaflet-popup-kibana').length).to.be(1);
+          });
+        });
+      });
+
+      describe('tooltipProximity method', function () {
+        it('should return true if feature is close enough to event latlng to display tooltip', function () {
```

Comment:
Given that some combo of `mapData`, `i`, `feature`, `point` and `map` and `zoom` are used in all of these tests, please move them to a `beforeEach()`